azkaban-common/src/main/c/execute-as-user.c

```diff
@@ -39,18 +39,23 @@ int INVALID_INPUT = 30;
 
 /*
  *  Change the real and effective user and group from super user to the specified user
- *  
+ *
  *  Adopted from:
  *  ./hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c
- *  
+ *
  */
 
-int change_user(uid_t user, gid_t group) {
+int change_user(char *username, uid_t user, gid_t group) {
```

Comment:
This actually is supposed to be a char* being passed. The initgroups() function below takes in a char* and a gid_t where the char* is the username of the user. When "uid" is passed from the main function, it actually is a char* - it's kind of hard to tell in github.